ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdCalculationService.java

```diff
@@ -81,6 +84,29 @@ public void setApplicationContext(@Nonnull ApplicationContext applicationContext
 		this.applicationContext = applicationContext;
 	}
 
+	PersistedInstanceIdStatisticsMBean collectStatistics(MirroredObject<?> mirroredObject) {
+		return getStatisticsForCollection(mirroredObject.getCollectionName());
+	}
+
+	// Initializes readyForNumberOfPartitions for collections that are already calculated
+	void initializeStatistics() {
+		Set<Integer> numberOfPartitionsToCalculate = getNumberOfPartitionsToCalculate();
+		getCollectionsWithPersistInstanceIdEnabled().forEach(collectionName -> {
+			try {
+				PersistedInstanceIdStatistics statistics = getStatisticsForCollection(collectionName);
+				numberOfPartitionsToCalculate.stream()
+						.filter(numberOfPartitions -> spaceMirror.getDocumentDb().getCollection(collectionName).getIndexes().anyMatch(isIndexForNumberOfPartitions(numberOfPartitions)))
+						.forEach(statistics::addReadyForNumberOfPartitions);
+			} catch (Exception e) {
+				log.warn("Unable to initialize persisted instance id statistics for collection {}", collectionName, e);
+			}
+		});
+	}
+
+	private PersistedInstanceIdStatistics getStatisticsForCollection(String collectionName) {
+		return jobStatistics.computeIfAbsent(collectionName, x -> new PersistedInstanceIdStatistics());
+	}
+
 	public boolean collectionNeedsCalculation(String collectionName) {
```

Comment:
It's not part of the scope of this MR, but is the result of this method really correct when also using `nextNumberOfInstances`? `noneMatch(isIndexForAny..(set))` is hard to parse for my old-person-brain. 

Here is a test that exemplifies it, but I have probably misunderstood it:
```java
@Test
public void test() throws Exception {
	var mongoDbFactory = mirrorEnvironment.getMongoTemplate().getMongoDbFactory();
	var collectionName = TEST_SPACE_OBJECT.collectionName();
	mongoDbFactory.getDb().getCollection(collectionName)
			.createIndex(new Document(getInstanceIdFieldName(1), 1));

	var testSpaceMirrorFactory = new TestSpaceMirrorFactory(mongoDbFactory);
	testSpaceMirrorFactory.setNextNumberOfInstances(9);

	try (var endpoint = (YmerSpaceSynchronizationEndpoint) testSpaceMirrorFactory.createSpaceSynchronizationEndpoint()) {
		var target = endpoint.getPersistedInstanceIdCalculationService();
		execute(
				() -> assertTrue(target.collectionNeedsCalculation(collectionName)),
				new SystemProperties("cluster.partitions", "1")
		);
	}
}
```

Reply:
No, you're right as your test exemplifies 😱  Thanks for finding this! This testcase also seems to be missing...
As you say, this is a part of an earlier PR (https://github.com/AvanzaBank/ymer/pull/66), but as that one is closed I'll fix it here.

This method was not working when multiple number of partitions were used. The result would have been that the job sometimes wouldn't have started correctly (if it already was calculated for the current number of partitions) 
 when a next number of instances is set.

I have pushed a fix for this bug now.

Reply:
I decided to remove the bugfix from this issue (as it's not really part of this PR), and open a separate PR: https://github.com/AvanzaBank/ymer/pull/69

So if you are done with the review of this issue, can you mark it as ready? 🙏